libs/chonky/src/lib.rs

```diff
@@ -1,26 +1,106 @@
 #![doc = include_str!("../README.md")]
 
-/// Adds two numbers together
-///
-/// # Example
+use error_stack::Report;
+use image::DynamicImage;
+use pdfium_render::prelude::*;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ChonkyError {
+    #[error("parsing error in pdf")]
+    ReadPdf,
+    #[error("pdfium error")]
+    Pdfium,
+    #[error("write error to system")]
+    Write,
+    #[error("Issues with CLI input")]
+    Arguments,
+}
+
+/// Function to read the pdf
 ///
-/// ```rust
-/// use chonky::add;
+/// # Errors
+/// Will return `Err` if `filename` does not exist or the user does not have
+/// permission to read it.
+pub fn load_pdf<'a>(
+    pdfium: &'a Pdfium,
+    file_path: &str,
+) -> Result<PdfDocument<'a>, Report<PdfiumError>> {
+    let pdf = pdfium.load_pdf_from_file(file_path, None)?;
+    Ok(pdf)
+}
+
+/// Takes in a pdf document and returns a vector list where each page
+/// is processed into a raw image that can be later converted to any image format
 ///
-/// assert_eq!(add(1, 3), 4);
-/// ```
-#[must_use]
-pub const fn add(left: u64, right: u64) -> u64 {
-    left + right
+/// # Errors
+/// Return an `PdfiumError::ImageError` if there was a problem with the image processing operation,
+/// occurs when image cannot be encoded into specific image format
+pub fn pdf_to_images(pdf: &PdfDocument) -> Result<Vec<DynamicImage>, Report<PdfiumError>> {
+    let mut images: Vec<DynamicImage> = Vec::new();
+
+    for page in pdf.pages().iter() {
+        let resolution_width = 1000; //may adjust resolution depending on need
```

Comment:
`rustfmt` for some reason sometimes has issues formatting the file when appending the comment. Also, when the comment gets to long it's likely that it's either not really readable or (worse) it's breaking into the next line, which could be read as a comment for the line after. This is clearer (and more consistent with the rest of the code):
```suggestion
        // may adjust resolution depending on the need
        let resolution_width = 1000;
```